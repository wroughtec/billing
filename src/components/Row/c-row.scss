.c-row__item {
  display: flex;
  min-height: 4rem;
  padding: 1rem;
  overflow: hidden;

  align-items: center;

  &:focus {
    outline: none;
  }
}

.c-row__obj--left {
  width: 4rem;
  margin-right: 1rem;
}

.c-row__obj--right {
  padding-left: 1rem;
  margin-right: 1rem;
  margin-left: auto;
}

.c-row__img {
  display: block;
  max-width: 100%;
}

.c-row__title {
  display: block;
  margin-top: 0;
  margin-bottom: 0;
}

.c-row__category {
  display: block;

  color: var(--infoColour);
}

.c-row__btn {
  display: inline-block;
  padding: .5rem .75rem;

  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: var(--btnText);
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;

  user-select: none;

  background-color: var(--btnBackground);
  border: 1px solid transparent;
  border-color: var(--btnBackground);
  border-radius: .25rem;

  transition: color var(--animationSpeed), background-color var(--animationSpeed), border-color var(--animationSpeed),
    box-shadow var(--animationSpeed);

  &:focus {
    text-decoration: none;

    outline: 0;
    box-shadow: var(--focusBoxShadow);
  }

  &:hover {
    color: var(--btnText);

    background-color: var(--btnHoverBackground);
    border-color: var(--btnHoverBackground);
  }

  &:not(:disabled):not(.disabled) {
    cursor: pointer;
  }

  &:not(:disabled):not(.disabled):active {
    color: var(--btnText);

    background-color: var(--btnHoverBackground);
    border-color: var(--btnHoverBackground);
  }
}
