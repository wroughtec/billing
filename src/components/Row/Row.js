// @flow
import React, { Component } from 'react';
import { Transactions } from '../Transactions/Transactions';
import './c-row.scss';

type Props = {
  data: BillsType
};

type State = {
  imgLoaded: boolean,
  openTransaction: boolean
};

export class Row extends Component<Props, State> {
  state = {
    imgError: false,
    openTransaction: false
  };

  handleImageError = () => {
    this.setState({
      imgError: true
    });
  };

  handleClick = (event: SyntheticEvent<HTMLButtonElement>) => {
    event.stopPropagation();

    const {
      data: { isBill, id },
      store
    } = this.props;

    return store.updateBill(id, !isBill);
  };

  mapCategories = (id: number) => {
    const { store } = this.props;
    let categoryName = null;
    if (store.categories && store.categories.length) {
      const category = store.categories.find(item => id === item.id);
      categoryName = category.name.toUpperCase();
    }

    return categoryName;
  };

  handleOpeningTransactions = () => {
    const { openTransaction } = this.state;

    this.setState({
      openTransaction: !openTransaction
    });
  };

  render() {
    const {
        data: { name, iconUrl, isBill, categoryId, transactions }
      } = this.props,
      { imgError, openTransaction } = this.state,
      btnLabel = isBill ? 'Remove bill' : 'Add as bill',
      categoryName = this.mapCategories(categoryId),
      transactionNumber = transactions.length;

    let img = null;

    if (!imgError && iconUrl) {
      img = <img onError={this.handleImageError} src={iconUrl} alt={name} className="c-row__img" />;
    }

    return (
      <div className="c-row">
        <div
          className="c-row__item"
          onClick={this.handleOpeningTransactions}
          role="button"
          onKeyPress={this.handleOpeningTransactions}
          tabIndex="0"
        >
          <div className="c-row__obj c-row__obj--left">{img}</div>
          <div className="c-row__body">
            <h4 className="c-row__title">{name}</h4>
            {categoryName && <small className="c-row__category">{categoryName}</small>}
            <small className="c-row__trans">Number of transactions: {transactionNumber}</small>
          </div>
          <div className="c-row__obj c-row__obj--right">
            <button className="c-row__btn" onClick={this.handleClick}>
              {btnLabel}
            </button>
          </div>
        </div>
        {openTransaction && <Transactions data={transactions} />}
      </div>
    );
  }
}
