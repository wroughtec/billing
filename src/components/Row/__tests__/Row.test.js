import React from 'react';
import { shallow, mount } from 'enzyme';
import { Row } from '../Row';
import db from '../../../__mocks__/db.json';

describe('Row', () => {
  let store;

  beforeEach(() => {
    store = {
      categories: db.categories,
      updateBill: jest.fn()
    };
  });

  const generateProps = (isBill = false) => {
    const bill = db.bills[0];
    bill.isBill = isBill;

    return bill;
  };

  test('Renders as a bill', () => {
    const data = generateProps(true);
    const component = shallow(<Row data={data} store={store} />);

    expect(component).toMatchSnapshot();
  });

  test('Renders as a transaction', () => {
    const data = generateProps();
    const component = shallow(<Row data={data} store={store} />);

    expect(component).toMatchSnapshot();
  });

  test('update event is fired', () => {
    const data = generateProps();
    const component = shallow(<Row data={data} store={store} />);
    component.find('button').simulate('click', { stopPropagation: () => undefined });

    expect(store.updateBill).toBeCalled();
  });

  test('open transaction event is fired', () => {
    const data = generateProps();
    const component = mount(<Row data={data} store={store} />);
    expect(component.find('.c-transactions').length).toBe(0);

    component.find('.c-row__item').simulate('click');

    expect(component.find('.c-transactions').length).toBe(1);
  });
});
