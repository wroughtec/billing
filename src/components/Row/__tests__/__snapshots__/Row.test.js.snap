// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Renders as a bill 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row
    data={
      Object {
        "categoryId": 1,
        "iconUrl": "https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg",
        "id": "5a5caa1efe33900100fd8ed5",
        "isBill": true,
        "name": "Vodafone",
        "transactions": Array [
          Object {
            "amount": 12.34,
            "date": "2018-01-13",
            "id": 36,
          },
          Object {
            "amount": 14.34,
            "date": "2018-02-13",
            "id": 37,
          },
          Object {
            "amount": 15.54,
            "date": "2018-03-13",
            "id": 38,
          },
          Object {
            "amount": 11.34,
            "date": "2018-04-13",
            "id": 39,
          },
          Object {
            "amount": 18.99,
            "date": "2018-05-13",
            "id": 40,
          },
        ],
      }
    }
    store={
      Object {
        "categories": Array [
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/income.png",
            "id": 1,
            "name": "Phone",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/entertainment.png",
            "id": 2,
            "name": "TV",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/bank_charges.png",
            "id": 3,
            "name": "Bank Charges",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/transport.png",
            "id": 4,
            "name": "Transport",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/groceries.png",
            "id": 5,
            "name": "Groceries",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/eating_out.png",
            "id": 6,
            "name": "Eating Out",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/shopping.png",
            "id": 7,
            "name": "Shopping",
          },
        ],
        "updateBill": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-row__item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="c-row__obj c-row__obj--left"
          >
            <img
              alt="Vodafone"
              className="c-row__img"
              onError={[Function]}
              src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
            />
          </div>
          <div
            className="c-row__body"
          >
            <h4
              className="c-row__title"
            >
              Vodafone
            </h4>
            <small
              className="c-row__category"
            >
              PHONE
            </small>
            <small
              className="c-row__trans"
            >
              Number of transactions: 
              5
            </small>
          </div>
          <div
            className="c-row__obj c-row__obj--right"
          >
            <button
              className="c-row__btn"
              onClick={[Function]}
            >
              Remove bill
            </button>
          </div>
        </div>,
        false,
      ],
      "className": "c-row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c-row__obj c-row__obj--left"
            >
              <img
                alt="Vodafone"
                className="c-row__img"
                onError={[Function]}
                src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
              />
            </div>,
            <div
              className="c-row__body"
            >
              <h4
                className="c-row__title"
              >
                Vodafone
              </h4>
              <small
                className="c-row__category"
              >
                PHONE
              </small>
              <small
                className="c-row__trans"
              >
                Number of transactions: 
                5
              </small>
            </div>,
            <div
              className="c-row__obj c-row__obj--right"
            >
              <button
                className="c-row__btn"
                onClick={[Function]}
              >
                Remove bill
              </button>
            </div>,
          ],
          "className": "c-row__item",
          "onClick": [Function],
          "onKeyPress": [Function],
          "role": "button",
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Vodafone"
                className="c-row__img"
                onError={[Function]}
                src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
              />,
              "className": "c-row__obj c-row__obj--left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Vodafone",
                "className": "c-row__img",
                "onError": [Function],
                "src": "https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="c-row__title"
                >
                  Vodafone
                </h4>,
                <small
                  className="c-row__category"
                >
                  PHONE
                </small>,
                <small
                  className="c-row__trans"
                >
                  Number of transactions: 
                  5
                </small>,
              ],
              "className": "c-row__body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vodafone",
                  "className": "c-row__title",
                },
                "ref": null,
                "rendered": "Vodafone",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "PHONE",
                  "className": "c-row__category",
                },
                "ref": null,
                "rendered": "PHONE",
                "type": "small",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Number of transactions: ",
                    5,
                  ],
                  "className": "c-row__trans",
                },
                "ref": null,
                "rendered": Array [
                  "Number of transactions: ",
                  5,
                ],
                "type": "small",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="c-row__btn"
                onClick={[Function]}
              >
                Remove bill
              </button>,
              "className": "c-row__obj c-row__obj--right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove bill",
                "className": "c-row__btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Remove bill",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-row__item"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="c-row__obj c-row__obj--left"
            >
              <img
                alt="Vodafone"
                className="c-row__img"
                onError={[Function]}
                src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
              />
            </div>
            <div
              className="c-row__body"
            >
              <h4
                className="c-row__title"
              >
                Vodafone
              </h4>
              <small
                className="c-row__category"
              >
                PHONE
              </small>
              <small
                className="c-row__trans"
              >
                Number of transactions: 
                5
              </small>
            </div>
            <div
              className="c-row__obj c-row__obj--right"
            >
              <button
                className="c-row__btn"
                onClick={[Function]}
              >
                Remove bill
              </button>
            </div>
          </div>,
          false,
        ],
        "className": "c-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c-row__obj c-row__obj--left"
              >
                <img
                  alt="Vodafone"
                  className="c-row__img"
                  onError={[Function]}
                  src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
                />
              </div>,
              <div
                className="c-row__body"
              >
                <h4
                  className="c-row__title"
                >
                  Vodafone
                </h4>
                <small
                  className="c-row__category"
                >
                  PHONE
                </small>
                <small
                  className="c-row__trans"
                >
                  Number of transactions: 
                  5
                </small>
              </div>,
              <div
                className="c-row__obj c-row__obj--right"
              >
                <button
                  className="c-row__btn"
                  onClick={[Function]}
                >
                  Remove bill
                </button>
              </div>,
            ],
            "className": "c-row__item",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Vodafone"
                  className="c-row__img"
                  onError={[Function]}
                  src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
                />,
                "className": "c-row__obj c-row__obj--left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Vodafone",
                  "className": "c-row__img",
                  "onError": [Function],
                  "src": "https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="c-row__title"
                  >
                    Vodafone
                  </h4>,
                  <small
                    className="c-row__category"
                  >
                    PHONE
                  </small>,
                  <small
                    className="c-row__trans"
                  >
                    Number of transactions: 
                    5
                  </small>,
                ],
                "className": "c-row__body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vodafone",
                    "className": "c-row__title",
                  },
                  "ref": null,
                  "rendered": "Vodafone",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "PHONE",
                    "className": "c-row__category",
                  },
                  "ref": null,
                  "rendered": "PHONE",
                  "type": "small",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Number of transactions: ",
                      5,
                    ],
                    "className": "c-row__trans",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Number of transactions: ",
                    5,
                  ],
                  "type": "small",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="c-row__btn"
                  onClick={[Function]}
                >
                  Remove bill
                </button>,
                "className": "c-row__obj c-row__obj--right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Remove bill",
                  "className": "c-row__btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Remove bill",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Row Renders as a transaction 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row
    data={
      Object {
        "categoryId": 1,
        "iconUrl": "https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg",
        "id": "5a5caa1efe33900100fd8ed5",
        "isBill": false,
        "name": "Vodafone",
        "transactions": Array [
          Object {
            "amount": 12.34,
            "date": "2018-01-13",
            "id": 36,
          },
          Object {
            "amount": 14.34,
            "date": "2018-02-13",
            "id": 37,
          },
          Object {
            "amount": 15.54,
            "date": "2018-03-13",
            "id": 38,
          },
          Object {
            "amount": 11.34,
            "date": "2018-04-13",
            "id": 39,
          },
          Object {
            "amount": 18.99,
            "date": "2018-05-13",
            "id": 40,
          },
        ],
      }
    }
    store={
      Object {
        "categories": Array [
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/income.png",
            "id": 1,
            "name": "Phone",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/entertainment.png",
            "id": 2,
            "name": "TV",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/bank_charges.png",
            "id": 3,
            "name": "Bank Charges",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/transport.png",
            "id": 4,
            "name": "Transport",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/groceries.png",
            "id": 5,
            "name": "Groceries",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/eating_out.png",
            "id": 6,
            "name": "Eating Out",
          },
          Object {
            "iconUrl": "https://s3.eu-west-2.amazonaws.com/cleoassets/images/category_icons/shopping.png",
            "id": 7,
            "name": "Shopping",
          },
        ],
        "updateBill": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-row__item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="c-row__obj c-row__obj--left"
          >
            <img
              alt="Vodafone"
              className="c-row__img"
              onError={[Function]}
              src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
            />
          </div>
          <div
            className="c-row__body"
          >
            <h4
              className="c-row__title"
            >
              Vodafone
            </h4>
            <small
              className="c-row__category"
            >
              PHONE
            </small>
            <small
              className="c-row__trans"
            >
              Number of transactions: 
              5
            </small>
          </div>
          <div
            className="c-row__obj c-row__obj--right"
          >
            <button
              className="c-row__btn"
              onClick={[Function]}
            >
              Add as bill
            </button>
          </div>
        </div>,
        false,
      ],
      "className": "c-row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c-row__obj c-row__obj--left"
            >
              <img
                alt="Vodafone"
                className="c-row__img"
                onError={[Function]}
                src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
              />
            </div>,
            <div
              className="c-row__body"
            >
              <h4
                className="c-row__title"
              >
                Vodafone
              </h4>
              <small
                className="c-row__category"
              >
                PHONE
              </small>
              <small
                className="c-row__trans"
              >
                Number of transactions: 
                5
              </small>
            </div>,
            <div
              className="c-row__obj c-row__obj--right"
            >
              <button
                className="c-row__btn"
                onClick={[Function]}
              >
                Add as bill
              </button>
            </div>,
          ],
          "className": "c-row__item",
          "onClick": [Function],
          "onKeyPress": [Function],
          "role": "button",
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Vodafone"
                className="c-row__img"
                onError={[Function]}
                src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
              />,
              "className": "c-row__obj c-row__obj--left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Vodafone",
                "className": "c-row__img",
                "onError": [Function],
                "src": "https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="c-row__title"
                >
                  Vodafone
                </h4>,
                <small
                  className="c-row__category"
                >
                  PHONE
                </small>,
                <small
                  className="c-row__trans"
                >
                  Number of transactions: 
                  5
                </small>,
              ],
              "className": "c-row__body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vodafone",
                  "className": "c-row__title",
                },
                "ref": null,
                "rendered": "Vodafone",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "PHONE",
                  "className": "c-row__category",
                },
                "ref": null,
                "rendered": "PHONE",
                "type": "small",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Number of transactions: ",
                    5,
                  ],
                  "className": "c-row__trans",
                },
                "ref": null,
                "rendered": Array [
                  "Number of transactions: ",
                  5,
                ],
                "type": "small",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="c-row__btn"
                onClick={[Function]}
              >
                Add as bill
              </button>,
              "className": "c-row__obj c-row__obj--right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add as bill",
                "className": "c-row__btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add as bill",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-row__item"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="c-row__obj c-row__obj--left"
            >
              <img
                alt="Vodafone"
                className="c-row__img"
                onError={[Function]}
                src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
              />
            </div>
            <div
              className="c-row__body"
            >
              <h4
                className="c-row__title"
              >
                Vodafone
              </h4>
              <small
                className="c-row__category"
              >
                PHONE
              </small>
              <small
                className="c-row__trans"
              >
                Number of transactions: 
                5
              </small>
            </div>
            <div
              className="c-row__obj c-row__obj--right"
            >
              <button
                className="c-row__btn"
                onClick={[Function]}
              >
                Add as bill
              </button>
            </div>
          </div>,
          false,
        ],
        "className": "c-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c-row__obj c-row__obj--left"
              >
                <img
                  alt="Vodafone"
                  className="c-row__img"
                  onError={[Function]}
                  src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
                />
              </div>,
              <div
                className="c-row__body"
              >
                <h4
                  className="c-row__title"
                >
                  Vodafone
                </h4>
                <small
                  className="c-row__category"
                >
                  PHONE
                </small>
                <small
                  className="c-row__trans"
                >
                  Number of transactions: 
                  5
                </small>
              </div>,
              <div
                className="c-row__obj c-row__obj--right"
              >
                <button
                  className="c-row__btn"
                  onClick={[Function]}
                >
                  Add as bill
                </button>
              </div>,
            ],
            "className": "c-row__item",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Vodafone"
                  className="c-row__img"
                  onError={[Function]}
                  src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
                />,
                "className": "c-row__obj c-row__obj--left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Vodafone",
                  "className": "c-row__img",
                  "onError": [Function],
                  "src": "https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="c-row__title"
                  >
                    Vodafone
                  </h4>,
                  <small
                    className="c-row__category"
                  >
                    PHONE
                  </small>,
                  <small
                    className="c-row__trans"
                  >
                    Number of transactions: 
                    5
                  </small>,
                ],
                "className": "c-row__body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vodafone",
                    "className": "c-row__title",
                  },
                  "ref": null,
                  "rendered": "Vodafone",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "PHONE",
                    "className": "c-row__category",
                  },
                  "ref": null,
                  "rendered": "PHONE",
                  "type": "small",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Number of transactions: ",
                      5,
                    ],
                    "className": "c-row__trans",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Number of transactions: ",
                    5,
                  ],
                  "type": "small",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="c-row__btn"
                  onClick={[Function]}
                >
                  Add as bill
                </button>,
                "className": "c-row__obj c-row__obj--right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add as bill",
                  "className": "c-row__btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add as bill",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
